api/src/lib.rs

```diff
@@ -80,14 +84,26 @@ impl PostieApi {
         println!("Successfully parsed postman environment!");
         Ok(())
     }
-    pub fn save_environment(input: Environment) -> Result<(), Box<dyn Error>> {
+    pub fn save_environment(_input: Environment) -> Result<(), Box<dyn Error>> {
         Ok(())
     }
-    pub fn save_collection(input: RequestCollection) -> Result<(), Box<dyn Error>> {
+    pub fn save_collection(_input: RequestCollection) -> Result<(), Box<dyn Error>> {
         Ok(())
     }
-    pub async fn make_request(input: HttpRequest) -> Result<Value, Box<dyn Error>> {
-        let client = Client::new();
+    pub fn substitute_variables_in_url(environment: &EnvironmentFile, raw_url: String) -> String {
+        println!("substituting env vars");
+        if let Some(values) = environment.clone().values {
+            let url = values.iter().fold(raw_url, |acc, env_value| {
+                acc.replace(&format!("{{{{{}}}}}", env_value.key), &env_value.value)
```

Comment:
To escape a `{` you need to `{{` I guess? So that ends up with this funkiness to escape a `{{}}` lol

Reply:
heh, funny. Reminds me of escaping `\` chars in javascript